### ISSUE TITLE
unable to install cantera development version "2.3.0a2" on rhel 6.5

### ISSUE THREAD

**Issue by pharthiphan** [Cantera/cantera#330]
unable to install cantera development version "2.3.0a2" on rhel 6.5. while compiling src/thermo/Nasa9PolyMultiTempRegion.cpp throws the below shown error.

/usr/include/c++/4.4.7/bits/vector.tcc(380): error: function "std::unique_ptr<_Tp, _Tp_Deleter>::unique_ptr(const std::unique_ptr<_Tp, _Tp_Deleter> &) [with _Tp=Cantera::Nasa9Poly1, _Tp_Deleter=std::default_deleteCantera::Nasa9Poly1]" (declared at line 214 of "/usr/include/c++/4.4.7/bits/unique_ptr.h") cannot be referenced -- it is a deleted function
              value_type __x_copy = __x;
                                    ^
          detected during:
            instantiation of "void std::vector<_Tp, _Alloc>::_M_fill_insert(std::vector<_Tp, _Alloc>::iterator, std::vector<_Tp, _Alloc>::size_type={size_t={unsigned long}}, const std::vector<_Tp, _Alloc>::value_type &) [with _Tp=std::unique_ptr<Cantera::Nasa9Poly1, std::default_delete<Cantera::Nasa9Poly1>>, _Alloc=std::allocator<std::unique_ptr<Cantera::Nasa9Poly1, std::default_delete<Cantera::Nasa9Poly1>>>]" at line 851 of "/usr/include/c++/4.4.7/bits/stl_vector.h"
            instantiation of "void std::vector<_Tp, _Alloc>::insert(std::vector<_Tp, _Alloc>::iterator, std::vector<_Tp, _Alloc>::size_type={size_t={unsigned long}}, const std::vector<_Tp, _Alloc>::value_type &) [with _Tp=std::unique_ptr<Cantera::Nasa9Poly1, std::default_delete<Cantera::Nasa9Poly1>>, _Alloc=std::allocator<std::unique_ptr<Cantera::Nasa9Poly1, std::default_delete<Cantera::Nasa9Poly1>>>]" at line 557 of "/usr/include/c++/4.4.7/bits/stl_vector.h"
            instantiation of "void std::vector<_Tp, _Alloc>::resize(std::vector<_Tp, _Alloc>::size_type={size_t={unsigned long}}, std::vector<_Tp, _Alloc>::value_type) [with _Tp=std::unique_ptr<Cantera::Nasa9Poly1, std::default_delete<Cantera::Nasa9Poly1>>, _Alloc=std::allocator<std::unique_ptr<Cantera::Nasa9Poly1, std::default_delete<Cantera::Nasa9Poly1>>>]" at line 62 of "src/thermo/Nasa9PolyMultiTempRegion.cpp"

icpc -o build/src/thermo/PDSS.os -c -std=c++0x -pthread -O3 -Wall -xHost -fp-model precise -Wcheck -fPIC -DNDEBUG -Iinclude -Iinclude/cantera/ext src/thermo/PDSS.cpp
/usr/include/c++/4.4.7/bits/stl_algobase.h(691): error: function "std::unique_ptr<_Tp, _Tp_Deleter>::operator=(const std::unique_ptr<_Tp, _Tp_Deleter> &) [with _Tp=Cantera::Nasa9Poly1, _Tp_Deleter=std::default_deleteCantera::Nasa9Poly1]" (declared at line 219 of "/usr/include/c++/4.4.7/bits/unique_ptr.h") cannot be referenced -- it is a deleted function
        *__first = __value;
                 ^
          detected during:
            instantiation of "__gnu_cxx::__enable_if<<expression>, void>::__type std::__fill_a(_ForwardIterator, _ForwardIterator, const _Tp &) [with _ForwardIterator=std::unique_ptr<Cantera::Nasa9Poly1, std::default_delete<Cantera::Nasa9Poly1>> *, _Tp=std::unique_ptr<Cantera::Nasa9Poly1, std::default_delete<Cantera::Nasa9Poly1>>]" at line 738
            instantiation of "void std::fill(_ForwardIterator, _ForwardIterator, const _Tp &) [with _ForwardIterator=std::unique_ptr<Cantera::Nasa9Poly1, std::default_delete<Cantera::Nasa9Poly1>> *, _Tp=std::unique_ptr<Cantera::Nasa9Poly1, std::default_delete<Cantera::Nasa9Poly1>>]" at line 393 of "/usr/include/c++/4.4.7/bits/vector.tcc"
            instantiation of "void std::vector<_Tp, _Alloc>::_M_fill_insert(std::vector<_Tp, _Alloc>::iterator, std::vector<_Tp, _Alloc>::size_type={size_t={unsigned long}}, const std::vector<_Tp, _Alloc>::value_type &) [with _Tp=std::unique_ptr<Cantera::Nasa9Poly1, std::default_delete<Cantera::Nasa9Poly1>>, _Alloc=std::allocator<std::unique_ptr<Cantera::Nasa9Poly1, std::default_delete<Cantera::Nasa9Poly1>>>]" at line 851 of "/usr/include/c++/4.4.7/bits/stl_vector.h"
            instantiation of "void std::vector<_Tp, _Alloc>::insert(std::vector<_Tp, _Alloc>::iterator, std::vector<_Tp, _Alloc>::size_type={size_t={unsigned long}}, const std::vector<_Tp, _Alloc>::value_type &) [with _Tp=std::unique_ptr<Cantera::Nasa9Poly1, std::default_delete<Cantera::Nasa9Poly1>>, _Alloc=std::allocator<std::unique_ptr<Cantera::Nasa9Poly1, std::default_delete<Cantera::Nasa9Poly1>>>]" at line 557 of "/usr/include/c++/4.4.7/bits/stl_vector.h"
            instantiation of "void std::vector<_Tp, _Alloc>::resize(std::vector<_Tp, _Alloc>::size_type={size_t={unsigned long}}, std::vector<_Tp, _Alloc>::value_type) [with _Tp=std::unique_ptr<Cantera::Nasa9Poly1, std::default_delete<Cantera::Nasa9Poly1>>, _Alloc=std::allocator<std::unique_ptr<Cantera::Nasa9Poly1, std::default_delete<Cantera::Nasa9Poly1>>>]" at line 62 of "src/thermo/Nasa9PolyMultiTempRegion.cpp"

/usr/include/c++/4.4.7/bits/stl_construct.h(73): error: function "std::unique_ptr<_Tp, _Tp_Deleter>::unique_ptr(const std::unique_ptr<_Tp, _Tp_Deleter> &) [with _Tp=Cantera::Nasa9Poly1, _Tp_Deleter=std::default_deleteCantera::Nasa9Poly1]" (declared at line 214 of "/usr/include/c++/4.4.7/bits/unique_ptr.h") cannot be referenced -- it is a deleted function
        ::new(static_cast<void*>(__p)) _T1(std::forward<_T2>(__value));
                                          ^
          detected during:
            instantiation of "void std::_Construct(_T1 *, _T2 &&) [with _T1=std::unique_ptr<Cantera::Nasa9Poly1, std::default_delete<Cantera::Nasa9Poly1>>, _T2=const std::unique_ptr<Cantera::Nasa9Poly1, std::default_delete<Cantera::Nasa9Poly1>> &]" at line 186 of "/usr/include/c++/4.4.7/bits/stl_uninitialized.h"
            instantiation of "void std::__uninitialized_fill_n<<unnamed>>::uninitialized_fill_n(_ForwardIterator, _Size, const _Tp &) [with <unnamed>=false, _ForwardIterator=std::unique_ptr<Cantera::Nasa9Poly1, std::default_delete<Cantera::Nasa9Poly1>> *, _Size=unsigned long, _Tp=std::unique_ptr<Cantera::Nasa9Poly1, std::default_delete<Cantera::Nasa9Poly1>>]" at line 223 of "/usr/include/c++/4.4.7/bits/stl_uninitialized.h"
            instantiation of "void std::uninitialized_fill_n(_ForwardIterator, _Size, const _Tp &) [with _ForwardIterator=std::unique_ptr<Cantera::Nasa9Poly1, std::default_delete<Cantera::Nasa9Poly1>> *, _Size=unsigned long, _Tp=std::unique_ptr<Cantera::Nasa9Poly1, std::default_delete<Cantera::Nasa9Poly1>>]" at line 317 of "/usr/include/c++/4.4.7/bits/stl_uninitialized.h"
            instantiation of "void std::__uninitialized_fill_n_a(_ForwardIterator, _Size, const _Tp &, std::allocator<_Tp2> &) [with _ForwardIterator=std::unique_ptr<Cantera::Nasa9Poly1, std::default_delete<Cantera::Nasa9Poly1>> *, _Size=unsigned long, _Tp=std::unique_ptr<Cantera::Nasa9Poly1, std::default_delete<Cantera::Nasa9Poly1>>, _Tp2=std::unique_ptr<Cantera::Nasa9Poly1, std::default_delete<Cantera::Nasa9Poly1>>]" at line 400 of "/usr/include/c++/4.4.7/bits/vector.tcc"
            instantiation of "void std::vector<_Tp, _Alloc>::_M_fill_insert(std::vector<_Tp, _Alloc>::iterator, std::vector<_Tp, _Alloc>::size_type={size_t={unsigned long}}, const std::vector<_Tp, _Alloc>::value_type &) [with _Tp=std::unique_ptr<Cantera::Nasa9Poly1, std::default_delete<Cantera::Nasa9Poly1>>, _Alloc=std::allocator<std::unique_ptr<Cantera::Nasa9Poly1, std::default_delete<Cantera::Nasa9Poly1>>>]" at line 851 of "/usr/include/c++/4.4.7/bits/stl_vector.h"
            instantiation of "void std::vector<_Tp, _Alloc>::insert(std::vector<_Tp, _Alloc>::iterator, std::vector<_Tp, _Alloc>::size_type={size_t={unsigned long}}, const std::vector<_Tp, _Alloc>::value_type &) [with _Tp=std::unique_ptr<Cantera::Nasa9Poly1, std::default_delete<Cantera::Nasa9Poly1>>, _Alloc=std::allocator<std::unique_ptr<Cantera::Nasa9Poly1, std::default_delete<Cantera::Nasa9Poly1>>>]" at line 557 of "/usr/include/c++/4.4.7/bits/stl_vector.h"
            instantiation of "void std::vector<_Tp, _Alloc>::resize(std::vector<_Tp, _Alloc>::size_type={size_t={unsigned long}}, std::vector<_Tp, _Alloc>::value_type) [with _Tp=std::unique_ptr<Cantera::Nasa9Poly1, std::default_delete<Cantera::Nasa9Poly1>>, _Alloc=std::allocator<std::unique_ptr<Cantera::Nasa9Poly1, std::default_delete<Cantera::Nasa9Poly1>>>]" at line 62 of "src/thermo/Nasa9PolyMultiTempRegion.cpp"

compilation aborted for src/thermo/Nasa9PolyMultiTempRegion.cpp (code 2)
scons: **\* [build/src/thermo/Nasa9PolyMultiTempRegion.os] Error 2
scons: building terminated because of errors.

this is how my cantera.conf looks

$ cat cantera.conf
CXX = 'icpc'
CC = 'icc'
prefix = '/root/cantera-install'
python_package = 'new'
python_prefix = 'USER'
python3_cmd = '/root/anaconda3/bin/python'
python3_prefix = 'USER'
f90_interface = 'y'
FORTRAN = 'ifort'
FORTRANFLAGS = '-O3 -xHost'
system_sundials = 'n'
blas_lapack_libs = 'mkl_rt,mkl_intel_lp64,mkl_core,mkl_intel_thread,iomp5'
blas_lapack_dir = '/gpfs1/opt/intel/icsxe/composer_xe_2013_sp1/mkl/lib/intel64'
env_vars = 'all'
cc_flags = ''
optimize_flags = '-O3 -Wall -xHost -fp-model precise'
debug = False

**Comment by pharthiphan** [comment]
$ **icpc -c -std=c++0x -pthread -O3 -Wall -xHost -fp-model precise -Wcheck -fPIC -DNDEBUG -Iinclude -Iinclude/cantera/ext src/thermo/Nasa9PolyMultiTempRegion.cpp**
/usr/include/c++/4.4.7/bits/vector.tcc(380): error: function "std::unique_ptr<_Tp, _Tp_Deleter>::unique_ptr(const std::unique_ptr<_Tp, _Tp_Deleter> &) [with _Tp=Cantera::Nasa9Poly1, _Tp_Deleter=std::default_deleteCantera::Nasa9Poly1]" (declared at line 214 of "/usr/include/c++/4.4.7/bits/unique_ptr.h") cannot be referenced -- it is a deleted function
              value_type __x_copy = __x;
                                    ^
          detected during:
            instantiation of "void std::vector<_Tp, _Alloc>::_M_fill_insert(std::vector<_Tp, _Alloc>::iterator, std::vector<_Tp, _Alloc>::size_type={size_t={unsigned long}}, const std::vector<_Tp, _Alloc>::value_type &) [with _Tp=std::unique_ptr<Cantera::Nasa9Poly1, std::default_delete<Cantera::Nasa9Poly1>>, _Alloc=std::allocator<std::unique_ptr<Cantera::Nasa9Poly1, std::default_delete<Cantera::Nasa9Poly1>>>]" at line 851 of "/usr/include/c++/4.4.7/bits/stl_vector.h"
            instantiation of "void std::vector<_Tp, _Alloc>::insert(std::vector<_Tp, _Alloc>::iterator, std::vector<_Tp, _Alloc>::size_type={size_t={unsigned long}}, const std::vector<_Tp, _Alloc>::value_type &) [with _Tp=std::unique_ptr<Cantera::Nasa9Poly1, std::default_delete<Cantera::Nasa9Poly1>>, _Alloc=std::allocator<std::unique_ptr<Cantera::Nasa9Poly1, std::default_delete<Cantera::Nasa9Poly1>>>]" at line 557 of "/usr/include/c++/4.4.7/bits/stl_vector.h"
            instantiation of "void std::vector<_Tp, _Alloc>::resize(std::vector<_Tp, _Alloc>::size_type={size_t={unsigned long}}, std::vector<_Tp, _Alloc>::value_type) [with _Tp=std::unique_ptr<Cantera::Nasa9Poly1, std::default_delete<Cantera::Nasa9Poly1>>, _Alloc=std::allocator<std::unique_ptr<Cantera::Nasa9Poly1, std::default_delete<Cantera::Nasa9Poly1>>>]" at line 62 of "src/thermo/Nasa9PolyMultiTempRegion.cpp"

/usr/include/c++/4.4.7/bits/stl_algobase.h(691): error: function "std::unique_ptr<_Tp, _Tp_Deleter>::operator=(const std::unique_ptr<_Tp, _Tp_Deleter> &) [with _Tp=Cantera::Nasa9Poly1, _Tp_Deleter=std::default_deleteCantera::Nasa9Poly1]" (declared at line 219 of "/usr/include/c++/4.4.7/bits/unique_ptr.h") cannot be referenced -- it is a deleted function
        *__first = __value;
                 ^
          detected during:
            instantiation of "__gnu_cxx::__enable_if<<expression>, void>::__type std::__fill_a(_ForwardIterator, _ForwardIterator, const _Tp &) [with _ForwardIterator=std::unique_ptr<Cantera::Nasa9Poly1, std::default_delete<Cantera::Nasa9Poly1>> *, _Tp=std::unique_ptr<Cantera::Nasa9Poly1, std::default_delete<Cantera::Nasa9Poly1>>]" at line 738
            instantiation of "void std::fill(_ForwardIterator, _ForwardIterator, const _Tp &) [with _ForwardIterator=std::unique_ptr<Cantera::Nasa9Poly1, std::default_delete<Cantera::Nasa9Poly1>> *, _Tp=std::unique_ptr<Cantera::Nasa9Poly1, std::default_delete<Cantera::Nasa9Poly1>>]" at line 393 of "/usr/include/c++/4.4.7/bits/vector.tcc"
            instantiation of "void std::vector<_Tp, _Alloc>::_M_fill_insert(std::vector<_Tp, _Alloc>::iterator, std::vector<_Tp, _Alloc>::size_type={size_t={unsigned long}}, const std::vector<_Tp, _Alloc>::value_type &) [with _Tp=std::unique_ptr<Cantera::Nasa9Poly1, std::default_delete<Cantera::Nasa9Poly1>>, _Alloc=std::allocator<std::unique_ptr<Cantera::Nasa9Poly1, std::default_delete<Cantera::Nasa9Poly1>>>]" at line 851 of "/usr/include/c++/4.4.7/bits/stl_vector.h"
            instantiation of "void std::vector<_Tp, _Alloc>::insert(std::vector<_Tp, _Alloc>::iterator, std::vector<_Tp, _Alloc>::size_type={size_t={unsigned long}}, const std::vector<_Tp, _Alloc>::value_type &) [with _Tp=std::unique_ptr<Cantera::Nasa9Poly1, std::default_delete<Cantera::Nasa9Poly1>>, _Alloc=std::allocator<std::unique_ptr<Cantera::Nasa9Poly1, std::default_delete<Cantera::Nasa9Poly1>>>]" at line 557 of "/usr/include/c++/4.4.7/bits/stl_vector.h"
            instantiation of "void std::vector<_Tp, _Alloc>::resize(std::vector<_Tp, _Alloc>::size_type={size_t={unsigned long}}, std::vector<_Tp, _Alloc>::value_type) [with _Tp=std::unique_ptr<Cantera::Nasa9Poly1, std::default_delete<Cantera::Nasa9Poly1>>, _Alloc=std::allocator<std::unique_ptr<Cantera::Nasa9Poly1, std::default_delete<Cantera::Nasa9Poly1>>>]" at line 62 of "src/thermo/Nasa9PolyMultiTempRegion.cpp"

/usr/include/c++/4.4.7/bits/stl_construct.h(73): error: function "std::unique_ptr<_Tp, _Tp_Deleter>::unique_ptr(const std::unique_ptr<_Tp, _Tp_Deleter> &) [with _Tp=Cantera::Nasa9Poly1, _Tp_Deleter=std::default_deleteCantera::Nasa9Poly1]" (declared at line 214 of "/usr/include/c++/4.4.7/bits/unique_ptr.h") cannot be referenced -- it is a deleted function
        ::new(static_cast<void*>(__p)) _T1(std::forward<_T2>(__value));
                                          ^
          detected during:
            instantiation of "void std::_Construct(_T1 *, _T2 &&) [with _T1=std::unique_ptr<Cantera::Nasa9Poly1, std::default_delete<Cantera::Nasa9Poly1>>, _T2=const std::unique_ptr<Cantera::Nasa9Poly1, std::default_delete<Cantera::Nasa9Poly1>> &]" at line 186 of "/usr/include/c++/4.4.7/bits/stl_uninitialized.h"
            instantiation of "void std::__uninitialized_fill_n<<unnamed>>::uninitialized_fill_n(_ForwardIterator, _Size, const _Tp &) [with <unnamed>=false, _ForwardIterator=std::unique_ptr<Cantera::Nasa9Poly1, std::default_delete<Cantera::Nasa9Poly1>> *, _Size=unsigned long, _Tp=std::unique_ptr<Cantera::Nasa9Poly1, std::default_delete<Cantera::Nasa9Poly1>>]" at line 223 of "/usr/include/c++/4.4.7/bits/stl_uninitialized.h"
            instantiation of "void std::uninitialized_fill_n(_ForwardIterator, _Size, const _Tp &) [with _ForwardIterator=std::unique_ptr<Cantera::Nasa9Poly1, std::default_delete<Cantera::Nasa9Poly1>> *, _Size=unsigned long, _Tp=std::unique_ptr<Cantera::Nasa9Poly1, std::default_delete<Cantera::Nasa9Poly1>>]" at line 317 of "/usr/include/c++/4.4.7/bits/stl_uninitialized.h"
            instantiation of "void std::__uninitialized_fill_n_a(_ForwardIterator, _Size, const _Tp &, std::allocator<_Tp2> &) [with _ForwardIterator=std::unique_ptr<Cantera::Nasa9Poly1, std::default_delete<Cantera::Nasa9Poly1>> *, _Size=unsigned long, _Tp=std::unique_ptr<Cantera::Nasa9Poly1, std::default_delete<Cantera::Nasa9Poly1>>, _Tp2=std::unique_ptr<Cantera::Nasa9Poly1, std::default_delete<Cantera::Nasa9Poly1>>]" at line 400 of "/usr/include/c++/4.4.7/bits/vector.tcc"
            instantiation of "void std::vector<_Tp, _Alloc>::_M_fill_insert(std::vector<_Tp, _Alloc>::iterator, std::vector<_Tp, _Alloc>::size_type={size_t={unsigned long}}, const std::vector<_Tp, _Alloc>::value_type &) [with _Tp=std::unique_ptr<Cantera::Nasa9Poly1, std::default_delete<Cantera::Nasa9Poly1>>, _Alloc=std::allocator<std::unique_ptr<Cantera::Nasa9Poly1, std::default_delete<Cantera::Nasa9Poly1>>>]" at line 851 of "/usr/include/c++/4.4.7/bits/stl_vector.h"
            instantiation of "void std::vector<_Tp, _Alloc>::insert(std::vector<_Tp, _Alloc>::iterator, std::vector<_Tp, _Alloc>::size_type={size_t={unsigned long}}, const std::vector<_Tp, _Alloc>::value_type &) [with _Tp=std::unique_ptr<Cantera::Nasa9Poly1, std::default_delete<Cantera::Nasa9Poly1>>, _Alloc=std::allocator<std::unique_ptr<Cantera::Nasa9Poly1, std::default_delete<Cantera::Nasa9Poly1>>>]" at line 557 of "/usr/include/c++/4.4.7/bits/stl_vector.h"
            instantiation of "void std::vector<_Tp, _Alloc>::resize(std::vector<_Tp, _Alloc>::size_type={size_t={unsigned long}}, std::vector<_Tp, _Alloc>::value_type) [with _Tp=std::unique_ptr<Cantera::Nasa9Poly1, std::default_delete<Cantera::Nasa9Poly1>>, _Alloc=std::allocator<std::unique_ptr<Cantera::Nasa9Poly1, std::default_delete<Cantera::Nasa9Poly1>>>]" at line 62 of "src/thermo/Nasa9PolyMultiTempRegion.cpp"

compilation aborted for src/thermo/Nasa9PolyMultiTempRegion.cpp (code 2)
[root@iitmlogin3 cantera]#

**Comment by speth** [comment]
Cantera 2.3.x requires GCC 4.6 or newer, while you appear to be using GCC 4.4. Note that even though you are using ICPC, you still need GCC because the Intel compiler uses the C++ standard library from there.

I don't know what version of the Intel compiler you are using, but you need to be using 14.0 or newer to compile the development version of Cantera.

**Comment by pharthiphan** [comment]
Hi Ray,

I was using intel version 13.0 and newer version is not available. so i am trying to install with gcc 4.8

[lg588710@bng1grcdcqdr47 cantera-master]$ rpm -qa | grep gcc
libgcc-4.8.5-4.el7.x86_64
gcc-4.8.5-4.el7.x86_64
gcc-gfortran-4.8.5-4.el7.x86_64
gcc-c++-4.8.5-4.el7.x86_64

and i am getitng the following error while doing "scons install", please guide

error: Invalid distribution name or version syntax: Cantera-minimal--2.3.0a2
scons: **\* [interfaces/python_minimal/dummy] Error 1
scons: building terminated because of errors.

**Comment by pharthiphan** [comment]
am i doing something wrong, please see the following lines from stdout while doing "scons build -j4"

[lg588710@bng1grcdcqdr47 cantera-master]$ scons build -j4
scons: Reading SConscript files ...
INFO: SCons is using the following Python interpreter: /usr/bin/python
Configuration variables read from 'cantera.conf' and command line:
    prefix = '/home/lg588710/cantera-install'

Checking for C++ header file cmath... yes
Checking for C++ header file gtest/gtest.h... no
Checking for C header file sys/times.h... yes
Checking for C header file unistd.h... yes
Checking for CVodeCreate(CV_BDF, CV_NEWTON) in C++ library sundials_cvodes... no
Checking for double x; log(x) in C library None... no
INFO: Using private installation of Sundials version 2.6.
Checking for C library gfortran... yes
INFO: Using 'gfortran' to build the Fortran 90 interface
WARNING: Cython not found or incompatible version: Found 0.0.0 but 0.19 or newer is required.
INFO: Not building the Python 3 package because the Python 3 interpreter 'python3' could not be found.
scons: done reading SConscript files.
scons: Building targets ...

**Comment by pharthiphan** [comment]
**scons test**

---

**\*    Testing Summary    ***

---

Tests passed: 175
Up-to-date tests skipped: 0
Tests failed: 0

---

scons: done building targets.
r]$ 

$ **scons install**
scons: Reading SConscript files ...
INFO: SCons is using the following Python interpreter: /usr/bin/python
Configuration variables read from 'cantera.conf' and command line:
    prefix = '/home/lg588710/cantera-install'

Checking for C++ header file cmath... (cached) yes
Checking for C++ header file gtest/gtest.h... (cached) no
Checking for C header file sys/times.h... (cached) yes
Checking for C header file unistd.h... (cached) yes
Checking for CVodeCreate(CV_BDF, CV_NEWTON) in C++ library sundials_cvodes... (cached) no
Checking for double x; log(x) in C library None... (cached) no
INFO: Using private installation of Sundials version 2.6.
Checking for C library gfortran... (cached) yes
INFO: Using 'gfortran' to build the Fortran 90 interface
WARNING: Cython not found or incompatible version: Found 0.0.0 but 0.19 or newer is required.
INFO: Not building the Python 3 package because the Python 3 interpreter 'python3' could not be found.
scons: done reading SConscript files.
scons: Building targets ...
ConfigBuilder(["include/cantera/base/config.h"], ["include/cantera/base/config.h.in"])
Generating include/cantera/base/config.h with the following settings:
    CANTERA_SHORT_VERSION               "2.3"
    CANTERA_VERSION                     "2.3.0a2"
    FTN_TRAILING_UNDERSCORE             1
    LAPACK_FTN_STRING_LEN_AT_END        1
    LAPACK_FTN_TRAILING_UNDERSCORE      1
    LAPACK_NAMES_LOWERCASE              1
    SUNDIALS_USE_LAPACK                 0
    SUNDIALS_VERSION                    26
ConfigBuilder(["include/cantera/ext/sundials/sundials_config.h"], ["ext/sundials_config.h.in"])
Generating include/cantera/ext/sundials/sundials_config.h with the following settings:
    SUNDIALS_USE_GENERIC_MATH           1
    SUNDIALS_BLAS_LAPACK                _undefined_
cd interfaces/python_minimal && "/usr/bin/python" setup.py build --build-lib=../../build/python2 install --user --record=../../build/python2-installed-files.txt --single-version-externally-managed
running build
running build_py
running install
running install_lib
copying ../../build/python2/cantera/**init**.pyc -> /home/lg588710/cantera-install/lib/python2.7/site-packages/cantera
copying ../../build/python2/cantera/ck2cti.pyc -> /home/lg588710/cantera-install/lib/python2.7/site-packages/cantera
copying ../../build/python2/cantera/ctml_writer.pyc -> /home/lg588710/cantera-install/lib/python2.7/site-packages/cantera
running install_egg_info
error: Invalid distribution name or version syntax: Cantera-minimal--2.3.0a2
scons: **\* [interfaces/python_minimal/dummy] Error 1
scons: building terminated because of errors.

**Comment by speth** [comment]
Can you try editing `interfaces/python_minimal/setup.py.in` and changing the third line from:

``` python
setup(name="Cantera (minimal)",
```

to

``` python
setup(name="Cantera_minimal",
```

**Comment by pharthiphan** [comment]
Thanks a lot :) it worked.I am able to compile on rhel 7.1 with gcc 4.8.5

I got to install cantera development version on rhel6.5 where we have gcc 4.4.7, so i have installed 4.8.5 by compiling gcc. exported the path,lib & include but i am getting the following error. please suggest

Checking for C++ header file cmath... no
ERROR: The C++ compiler is not correctly configured.

**Comment by speth** [comment]
What is the full output of the `scons build` command? What are the contents of the `config.log` file? Are you sure it's calling the correct copy of gcc and not the incompatible system version?

**Comment by pharthiphan** [comment]
**$ scons build**
scons: Reading SConscript files ...
INFO: SCons is using the following Python interpreter: /usr/bin/python
Configuration variables read from 'cantera.conf' and command line:

Checking for C++ header file cmath... no
ERROR: The C++ compiler is not correctly configured.
See 'config.log' for details.
**$ cat config.log**
file /home/lg588710/cantera-master/SConstruct,line 686:
        Configure(confdir = .sconf_temp)
scons: Configure: Checking for C++ header file cmath...
scons: Configure: ".sconf_temp/conftest_0.cpp" is up to date.
scons: Configure: The original builder output was:
  |.sconf_temp/conftest_0.cpp <-
  |  |
  |  |#include <cmath>
  |  |
  |  |
  |
g++ -o .sconf_temp/conftest_0.o -c -std=c++0x -pthread -O3 -Wno-inline -g -DNDEBUG .sconf_temp/conftest_0.cpp
/home/lg588710/gcc/gcc-4.8.5/libexec/gcc/x86_64-unknown-linux-gnu/4.8.5/cc1plus: error while loading shared libraries: libmpc.so.3: cannot open shared object file: No such file or directory
scons: Configure: no

but if i compile manually from linux prompt, it runs without any error!!!
*_$ g++ -o .sconf_temp/conftest_0.o -c -std=c++0x -pthread -O3 -Wno-inline -g -DNDEBUG *_.sconf_temp/conftest_0.cpp
$

**it is taking the gcc 4.8.5 which i have compile and build but it couldn't find the cmath** 

**gcc include path has been exported but still it couln't find cmath library**
export INCLUDE=/home/pharthiphan/gcc/gcc-4.8.5/include:$INCLUDE

$which gcc
~/gcc/gcc-4.8.5/bin/gcc
$ which g++
~/gcc/gcc-4.8.5/bin/g++
$ gcc --version
gcc (GCC) 4.8.5
Copyright (C) 2015 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

$ g++ --version
g++ (GCC) 4.8.5
Copyright (C) 2015 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

$

$ ls -lrt ~/gcc/gcc-4.8.5/include/c++/4.8.5/cmath
-rw-r--r-- 1 pharthiphan USERS 42445 Apr 14 11:27 /home/pharthiphan/gcc/gcc-4.8.5/include/c++/4.8.5/cmath
$

Please suggest!!

**Comment by speth** [comment]
The error is not that it can't find cmath, but this:

```
/home/lg588710/gcc/gcc-4.8.5/libexec/gcc/x86_64-unknown-linux-gnu/4.8.5/cc1plus: error while loading shared libraries: libmpc.so.3: cannot open shared object file: No such file or directory
```

I guess you're relying on `LD_LIBRARY_PATH` as well? By default, to ensure repeatable builds, SCons does not inherit the values of environment variables. If you need to tell SCons to pass through the environment variables by running `scons build` with the option `env_vars=all`.

**Comment by pharthiphan** [comment]
That helped thanks Ray. I am able install Cantera 2.3.0a2 by compiling gcc-4.8.5 for rhel 6.5

Thanks again!!!

**Comment by speth** [comment]
No problem.

**Comment by OssamaMannaa** [comment]
Dear all
I have been struggling with installing Cantera on Mac .Whenever i use Scons build i get the following error. Any help is really appreciated 

scons: Reading SConscript files ...
INFO: SCons is using the following Python interpreter: /opt/local/Library/Frameworks/Python.framework/Versions/2.7/Resources/Python.app/Contents/MacOS/Python
Configuration variables read from 'cantera.conf' and command line:
    prefix = '/Users/mannaaoa/cantera'
    python_package = 'full'
    python_cmd = '/usr/local/lib/python2.7/site-packages'
    python3_cmd = '/usr/local/lib/python3.4/site-packages'
    use_sundials = 'y'
    sundials_include = '/Users/mannaaoa/sundials/installdir/include'
    sundials_libdir = '/Users/mannaaoa/sundials/installdir/lib'

Checking for C++ header file cmath... yes
Checking for C header file sys/times.h... yes
Checking for C header file unistd.h... yes
Checking for C header file fenv.h... yes
Checking whether erf is declared... yes
Checking for ::isnan(1.0)... no
Checking for std::isnan(1.0)... yes
Checking for _isnan(1.0)... no
Checking for C++ header file boost/math/special_functions/erf.hpp... yes
Checking for std::shared_ptr<int> x... no
Checking for std::tr1::shared_ptr<int> x... yes
Checking for CVodeCreate(CV_BDF, CV_NEWTON) in C++ library sundials_cvodes... yes
Checking for double x; log(x) in C library None... yes
INFO: Using Sundials version 2.6.0.
Checking whether SUNDIALS_BLAS_LAPACK is declared... no
WARNING: External BLAS/LAPACK has been specified for Cantera but Sundials was built without this support.
Checking for C library gfortran... yes
INFO: Using 'gfortran' to build the Fortran 90 interface
ERROR: Cython not found or incompatible version: Found 0.0.0 but 0.19 or newer is required.

It seems that the issue with Cython. I suspect that Scons use  the Python interpreter from a wrong path but i do not know how to change it.I am not sure if i am right or wrong about my guess. I attached the config file

Thanks


[config.txt](https://github.com/Cantera/cantera/files/1177651/config.txt)
